client/src/legacy/GridField.js

```diff
@@ -181,30 +176,6 @@ $.entwine('ss', function($) {
         var colspan = cell.attr('colspan') ?? 1;
         cell.attr('colspan', Number(colspan) + 1);
       });
-      var $extraCell = $('<th class="extra" />');
-      $('.grid-field__filter-header th:last .action').each(function() {
-        $(this).detach();
-        $extraCell.append($(this));
-      });
-      $('.grid-field__filter-header').append($extraCell);
-    },
-
-    injectSearchButton: function(visible) {
-      const hasLegacyFilterHeader = this.find('.grid-field__filter-header').length > 0;
-      let content;
-      if (visible) {
-        content = '<span class="non-sortable"></span>';
-        this.addClass('show-filter').find('.grid-field__filter-header, .grid-field__search-holder').removeClass('grid-field__search-holder--hidden');
-        if (!hasLegacyFilterHeader) {
-          this.find(':button[name=showFilter]').hide();
-        }
-      } else {
-        content = '<button type="button" title="Open search and filter" name="showFilter" class="btn btn-secondary font-icon-search btn--no-text btn--icon-lg grid-field__filter-open"></button>';
-        this.removeClass('show-filter').find('.grid-field__filter-header, .grid-field__search-holder').addClass('grid-field__search-holder--hidden');
-      }
-      if (hasLegacyFilterHeader) {
-        this.find('.sortable-header th:last').html(content);
-      }
```

Comment:
Most of this is for the legacy filter that was removed in CMS 5 - anything related to the current filter isn't gridfield-specific so doesn't belong here.